web-client/src/presenter/actions/TrialSession/computeTrialSessionFormDataAction.js

```diff
@@ -67,26 +71,39 @@ const compute24HrTime = ({ extension, hours, minutes }) => {
 export const computeTrialSessionFormDataAction = ({ get, props, store }) => {
   const form = get(state.form);
 
-  const { term, termYear } = computeTerm({
-    month: form.startDateMonth,
-    year: form.startDateYear,
-  });
-  store.set(state.form.term, term);
-  store.set(state.form.termYear, termYear);
+  computeTermAndUpdateState(
+    {
+      month: form.startDateMonth,
+      year: form.startDateYear,
+    },
+    store,
+  );
+
+  compute24HrTimeAndUpdateState(
+    {
+      extension: form.startTimeExtension,
+      hours: form.startTimeHours,
+      minutes: form.startTimeMinutes,
+    },
+    store,
+  );
 
-  const startTime = compute24HrTime({
-    extension: form.startTimeExtension,
-    hours: form.startTimeHours,
-    minutes: form.startTimeMinutes,
-  });
-  store.set(state.form.startTime, startTime);
   if (props.key === 'judgeId') {
     store.set(state.form.judgeId, props.value.userId);
     store.set(state.form.judge, props.value);
   }
 
   if (props.key === 'trialClerkId') {
-    store.set(state.form.trialClerkId, props.value.userId);
-    store.set(state.form.trialClerk, props.value);
+    if (props.value) {
+      store.set(state.form.trialClerkId, props.value.userId);
+      if (props.value.userId !== 'Other') {
+        store.set(state.form.trialClerk, props.value);
+        store.set(state.form.alternateTrialClerkName, undefined);
+      }
+    } else {
+      store.set(state.form.alternateTrialClerkName, undefined);
```

Comment:
Nit: This comment is more for informational purposes, you don't really have to change anything but I think the method `store.unset(state.form.alternateTrialClerkName)` could also be used to achieve the same outcome!